plugins/uast-kotlin/src/org/jetbrains/uast/kotlin/declarations/KotlinUClass.kt

```diff
@@ -128,27 +128,37 @@ class KotlinConstructorUMethod(
                 else -> null
             }
         }
-        val initializers = ktClass?.getAnonymousInitializers().orEmpty()
-        if (delegationCall == null && initializers.isEmpty()) return@lz null
+        val bodyExpressions = getBodyExpressions()
+        if (delegationCall == null && bodyExpressions.isEmpty()) return@lz null
         KotlinUBlockExpression.KotlinLazyUBlockExpression(this) { uastParent ->
             SmartList<UExpression>().apply {
                 delegationCall?.let {
                     add(KotlinUFunctionCallExpression(it, uastParent))
                 }
                 val languagePlugin = uastParent.getLanguagePlugin()
-                initializers.forEach {
-                    addIfNotNull(languagePlugin.convertOpt(it.body, uastParent))
+                bodyExpressions.forEach {
+                    addIfNotNull(languagePlugin.convertOpt(it, uastParent))
                 }
             }
         }
     }
+
+    private fun getBodyExpressions(): List<KtExpression> {
+        if (isPrimary) return getInitializers()
+        val bodyExpression = (psi.kotlinOrigin as? KtFunction)?.bodyExpression ?: return emptyList()
```

Comment:
I think that, if a class has no primary constructor, the body of each secondary constructor should include the list of initializer blocks prepended to it. This will match the behavior of the JVM backend.

Reply:
but then it is ambiguity on who will be the parent of these elements when building tree from down-to-top

Reply:
and also if in `init` block there will be something to report then it will be reported number of secondary constructors times. So it is not a good idea to copy initializer blocks.

Reply:
Then maybe copy it only under the first secondary constructor?

Reply:
it will go against the truth, but I agree maybe it is better than just skip it... Ok lets go this way